### ISSUE TITLE
"inaccessible site" not flagged by axe

### ISSUE THREAD

**Issue by paulirish** [dequelabs/axe-core#1607]
I imagine y'all saw this blog post from last week, authored by @matuzo: https://www.matuzo.at/blog/building-the-most-inaccessible-site-possible-with-a-perfect-lighthouse-score/

Here's the last iteration as a standalone page: https://s.codepen.io/paulirish/debug/VOqyJX

axe only detects two problems on this page: 
![image](https://user-images.githubusercontent.com/39191/58949229-3bcd6b00-878c-11e9-8cf4-2aa918fd9967.png)

I am curious of the axe team's thoughts on the 9 inaccessible changes added to the test page and if axe should be detecting some of them.

**Comment by matuzo** [comment]
Hey!

Thanks for opening the issue @paulirish. I didn't do it myself because I wasn't sure if any of the things I did there should be flagged as errors or warnings. 


I guess there could be at least a warning for small font sizes.
```
body {
  font-size: 1px;
}
```

**Comment by WilcoFiers** [comment]
Jup, we saw the post :). Interesting read. I haven't quite figured out what we could take away from it. Never the less, worth while reminding the world that automated tools can't test everything, and if you try hard enough, there will always be ways to trick them.

@paulirish @matuzo do either have you have suggestions for rules that we could add to axe-core based on this article? The one for putting aria-hidden on stuff with focusable content we have as of axe-core 3.2. We probably need to do more with the `hidden` attribute. Outline is another one we can probably do more with, now that we've built in utilities to access the CSSOM. Anything else?

**Comment by dylanb** [comment]
The Web site is a bit artificial in that it is completely inaccessible to everyone, which is not a very realistic scenario but I understand the point it is making.

On the other hand, we specifically do not claim that axe-core finds all the possible accessibility issues. We try to find issues with 100% accuracy so that developers can rely on the results of axe-core.

PS: if you are interested in some of the things we are doing to address the stuff that cannot currently be done within axe-core, sign up for our[ axe-pro](https://www.deque.com/axe/axe-pro/) beta (coming soon)

@WilcoFiers could we close this issue and open up specific discussions on each individual technique?

**Comment by matuzo** [comment]
@WilcoFiers, as @dylanb said, most of the things I did were unrealistic. `cursor:none` for example. For the realistic things I'm not sure if anything should/could be done. Even `outline: none;` is fine if there's `:focus-visible` or a polyfill. There could be a warning, also if really small font sizes have been applied and maybe opacity: 0; on focusable content?

**Comment by dylanb** [comment]
opacity:0 on screen reader visible content is a technique that gets used for providing an alternative for complex graphics

opacity:0 on visible content could be used to provide state information (it is a bit of a weird technique for this), so the best we could do there is raise a needs review.

really small font sizes is not technically a WCAG violation (there is no minimum font size requirement), we could raise this as a best practice violation.
